Strategy11/formidable-forms
# Try to optimize FrmEntryMeta::get_ids_query

Fix https://github.com/Strategy11/formidable-pro/issues/5167

**Testing instructions**
This update changes how the database is queried when running many queries that check for entry IDs. This includes the `[frm-search]` functionality in Formidable Views.

You can view the database queries by activating the Query Monitor plugin.

**Before**
The query when searching for entries before included two `JOIN` statements. Getting the data from three tables is quite slow.

**After**
The slow `LEFT OUTER JOIN` that joins data from the fields table is now removed from the query when there is no required field data in the query. This changes the query when searching using `[frm-search]` so it only includes a single `JOIN`. The new query (when searching data for ~50k entries), dropped from ~1.5s to ~1.1s, saving approximately 0.4 seconds from the page load (which is quite significant).

**What to test for**
- Confirming the queries when searching are still accurate.
- Confirming there are no entry ID queries that are now broken.
- Confirming the queries when searching are more efficient.
